.clang-tidy

```diff
@@ -14,6 +14,16 @@ misc-*,\
 -misc-static-assert,\
 -misc-unused-parameters,\
 -misc-non-private-member-variables-in-classes,\
+readability-*,\
+-readability-function-size,\
+-readability-identifier-naming,\
+-readability-implicit-bool-conversion,\
+-readability-inconsistent-declaration-parameter-name,\
+-readability-isolate-declaration,\
+-readability-magic-numbers,\
+-readability-named-parameter,\
+-readability-qualified-auto,\
+-readability-uppercase-literal-suffix,\
```

Comment:
I looked through each one of the `readability-*` checks.  Most of them I agree with, with two exceptions:

1) The `readability-else-after-return` check should be removed, as there are many places in the codebase where this happens.
2) The `readability-redundant-member-init` check should be removed as well.  IIRC there are a few classes with many members that are initialized, and the initialization list matches the list of fields.  Explicitly calling the default constructor visually indicates that the coder did not forget about initializing that field.

Reply:
1. I can see that. I'll add it to the ignore list.
2. I don't agree on this. Calling the default constructor explicitly just adds unnecessary clutter and makes it harder to see actual issues in the code.

Reply:
I don't feel as strongly about 2) as the other points, but consider these two examples from the code.  Wouldn't they be flagged by the `readability-redundant-member-init` check?

```
struct DiyFp {
    DiyFp() : f(), e() {}
```
```
	main_hall_region()
		: mask(0), key(0), description(), action(0), lua_action()
	{}
```

Reply:
Yes, both cases would be flagged. In my opinion, that is correct since fixing it would remove code and non-existing code can not cause bugs.

Reply:
Ok, I'm willing to see what happens.